src/udev/udev-node.c

```diff
@@ -232,7 +229,42 @@ static int link_update(sd_device *dev, const char *slink, bool add) {
                                 r = -errno;
                 } while (r == -ENOENT);
 
-        return r;
+        for (i = 0; i < LINK_UPDATE_MAX_RETRIES; i++) {
+                struct stat st1, st2 = {};
+
+                r = stat(dirname, &st1);
+                if (r < 0 && errno != ENOENT)
+                        return -errno;
```

Comment:
hmm, so in contrast to what i said before, st1 needs to be initialized up top, otherwise if we really see ENOENT here, it will be uninitialized, and we'd thus compare uninitialized memory. sorry for the confusion!